src/domain/wallet-invoices/wallet-invoice-receiver.ts

```diff
@@ -1,53 +1,77 @@
-import { WalletCurrency, ZERO_BANK_FEE } from "@domain/shared"
+import { PriceRatio } from "@domain/payments"
+import { AmountCalculator, WalletCurrency, ZERO_BANK_FEE } from "@domain/shared"
+
+const calc = AmountCalculator()
 
 export const WalletInvoiceReceiver = async ({
   walletInvoice,
   receivedBtc,
+  feeBtc = ZERO_BANK_FEE.btcBankFee,
   usdFromBtc,
   usdFromBtcMidPrice,
 }: WalletInvoiceReceiverArgs): Promise<
   WalletInvoiceReceiver | DealerPriceServiceError | ValidationError
 > => {
-  const btcToCreditReceiver = receivedBtc
-
   if (walletInvoice.recipientWalletDescriptor.currency === WalletCurrency.Btc) {
-    const usdToCreditReceiver = await usdFromBtcMidPrice(btcToCreditReceiver)
-    if (usdToCreditReceiver instanceof Error) return usdToCreditReceiver
+    const receivedUsd = await usdFromBtcMidPrice(receivedBtc)
+    if (receivedUsd instanceof Error) return receivedUsd
+
+    const priceRatio = PriceRatio({ usd: receivedUsd, btc: receivedBtc })
+    if (priceRatio instanceof Error) return priceRatio
+    const bankFee = {
+      usdBankFee: priceRatio.convertFromBtcToCeil(feeBtc),
+      btcBankFee: feeBtc,
+    }
 
     return {
-      btcToCreditReceiver,
-      usdToCreditReceiver,
-      ...ZERO_BANK_FEE,
+      btcToCreditReceiver: calc.sub(receivedBtc, feeBtc),
+      usdToCreditReceiver: calc.sub(receivedUsd, bankFee.usdBankFee),
+      ...bankFee,
```

Comment:
does not worth desctructure 2 properties

Reply:
Like this? https://github.com/GaloyMoney/galoy/pull/2157/commits/67ab149edd4e558dc302b801d5257da70ebc9d04